api.swagger.yaml

```diff
@@ -1327,6 +2254,233 @@ components:
+            example: 01234567-89ab-cdef-1234-56789abcdef0
+          likes_count:
+            description: Count of likes on the post
+            type: integer
+            example: 5
+          comment_count:
+            description: Count of comments on the post
+            type: integer
+            example: 5
+          attachments_count:
+            description: Count of attachments on the post
+            type: integer
+            example: 5
+    post-schema-create:
+      allOf:
+      - $ref: '#/components/schemas/post-schema-base'
+      - type: object
+        properties:
+          parent_uuid:
+            description: This is the Invotra UUID of the parent of the post.
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+            example: 01234567-89ab-cdef-1234-56789abcdef0
+          author_uuid:
+            description: The uuid of the author
+            type: string
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+            example: 01234567-89ab-cdef-1234-56789abcdef0
+          group_uuid:
+            description: This is the Invotra UUID of the group that the post belongs to.
+            pattern: >-
+              ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+            example: 01234567-89ab-cdef-1234-56789abcdef0
+            type: string
+      required:
+        - description
+        - group_uuid
+        - category
+        - author_uuid
+    post-schema-base:
+      properties:
+        description:
+          description: The main body for the post.
+          type: string
+          example: 'Check out the latest #security news in this group!'
+        external_id:
+          description: This is used to store the post's external reference id
+          type: string
+          example: ABC123
+        category:
+          description: The category of a post
+          type: string
+          enum:
+            - Comment
+            - Idea
+            - News / Event
+            - Note
+            - Question
+            - To Do
+            - Problem / Risk
+            - Agreed
+            - Decision
+            - Done
+        attachments:
+          allOf:
+            - $ref: '#/components/schemas/uuid-list-schema'
+      type: object
+    post-report-schema-create:
+      properties:
+        reason:
+          description: Reason for flagging post as inappropriate
+          type: string
+          example: I find this offensive
+    file-schema:
+      type: object
+      properties:
+        filename:
+          description: This is the filename
+          type: string
+          example: wallpaper.jpeg
+        file_download:
+          description: The location of the file
+          type: string
+          example: 'file/1234/download'
```

Comment:
Wouldn't this be a full URL with a protocol and host etc?

Reply:
updated